dulwich/tests/test_porcelain.py

```diff
@@ -465,12 +465,12 @@ def test_simple(self):
         # Check that the target and source
         r_clone = Repo(clone_path)
 
+        self.assertEqual(r_clone[b'HEAD'].id, self.repo[refs_path].id)
```

Comment:
The test was failing at one point, and the failure message was a confusing `IndexError` from the `change = list(tree_changes(...))[0]` line because there were no changes.

The change in order is so we first check that the push happened by checking the `HEAD` changed, so that the failure message is more understandable.